### PR TITLE
Add React single-page sample app

### PR DESCRIPTION
**Pull request by Dickson-Mwendia**
## Purpose

*  This PR is for #DOCS-CODE-023 - React single-page app . It imports the sample from https://github.com/Azure-Samples/ms-identity-javascript-react-docs 

Cc: @dhruvchand @ckittel @henrymbuguakiarie @mmacy

### CODE REVIEW

File: react-single-page-app/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+This minimal React application demonstrates usage of the Microsoft Authentication Library for React (MSAL React) to:
+
+- Sign in Azure AD users (authentication)
+- Call a protected web API (authorization)
+- Sign out users
+
+This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+
+## Available scripts
+
```
Comment (ckittel) [inline react-single-page-app/README.md:12]: Can you add to the backlog that this needs to be updated to the new style?  Not a blocker for this import, but should be done soon.

File: react-single-page-app/src/index.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+// <ms_docref_import_modules>
+import { Configuration,  PublicClientApplication } from "@azure/msal-browser";
+import { MsalProvider, useMsal, useIsAuthenticated } from "@azure/msal-react";
+// </ms_docref_import_modules>
+
+// <ms_docref_configure_msal>
+const configuration: Configuration = {
+  auth: {
+      clientId: "034ff080-d093-433d-915c-4f12ae03f100"
```
Comment (ckittel) [inline react-single-page-app/src/index.js:12]: Can you please replace this with our standard placeholder text (seen in other examples)

File: react-single-page-app/src/index.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+const getGraphToken =  async (msalInstance, accounts) => {
+  const tokenRequest = {
+    scopes: ["User.Read", "openid", "profile"],
+    account: accounts
+  }
+  
+  try{
+    const {accessToken} = await msalInstance.acquireTokenSilent(tokenRequest);
+    return accessToken;
+  } catch (e) {
```
Comment (ckittel) [inline react-single-page-app/src/index.js:29]: Should this be only a specific exception?  This seems really broad for this specific followup code in the catch.

File: react-single-page-app/src/index.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  
+}
+// </ms_docref_get_graph_token>
+
+// <ms_docref_make_graph_call>
+const MICROSOFT_GRAPH_URL = "https://graph.microsoft.com/v1.0"
+
+const fetchUserDetails = async (msalInstance, accounts, setUserDetails) => {
+  const bearer = `Bearer ${await getGraphToken(msalInstance, accounts)}`;
+  const response = await fetch(`${MICROSOFT_GRAPH_URL}/me`, {
```
Comment (ckittel) [inline react-single-page-app/src/index.js:42]: Inline `https://graph.microsoft.com/v1.0` to simplify the code.  (This is learnings from other examples we have built)

File: react-single-page-app/src/index.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  }
+  
+}
+// </ms_docref_get_graph_token>
+
+// <ms_docref_make_graph_call>
+const MICROSOFT_GRAPH_URL = "https://graph.microsoft.com/v1.0"
+
+const fetchUserDetails = async (msalInstance, accounts, setUserDetails) => {
+  const bearer = `Bearer ${await getGraphToken(msalInstance, accounts)}`;
```
Comment (ckittel) [inline react-single-page-app/src/index.js:41]: Inline the code to get the graph token here in this function. (This is learnings/requirements from other examples we have built)

File: react-single-page-app/src/index.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+  }
+};
+
+const clientInstance = new PublicClientApplication(configuration);
+// </ms_docref_configure_msal>
+
+// <ms_docref_get_graph_token>
+const getGraphToken =  async (msalInstance, accounts) => {
+  const tokenRequest = {
+    scopes: ["User.Read", "openid", "profile"],
```
Comment (ckittel) [inline react-single-page-app/src/index.js:22]: Should only need `User.Read` -- the rest are automatic (and not able to be changed anyhow)

File: react-single-page-app/src/index.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    account: accounts
+  }
+  
+  try{
+    const {accessToken} = await msalInstance.acquireTokenSilent(tokenRequest);
+    return accessToken;
```
Comment (ckittel) [inline react-single-page-app/src/index.js:28]: Simplification.

```suggestion
    return await msalInstance.acquireTokenSilent(tokenRequest);
```

File: react-single-page-app/src/index.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+  } catch (e) {
+    const {accessToken} = await msalInstance.acquireTokenPopup(tokenRequest);
+    return accessToken;
```
Comment (ckittel) [inline react-single-page-app/src/index.js:31]: Simplification.
```suggestion
    return await msalInstance.acquireTokenPopup(tokenRequest);
```

File: react-single-page-app/src/index.js [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+  setUserDetails(response)
+}
+// </ms_docref_make_graph_call>
+
+const PageContent = () => {
+
+  const [userDetails, setUserDetails] = useState({});
+
+  // <ms_docref_use_msal_context>
+  const { instance, accounts } = useMsal();
```
Comment (ckittel) [inline react-single-page-app/src/index.js:58]: Suggestion, use a more meaningful name than `instance`.

**Review by ckittel — CHANGES_REQUESTED** [review]
Glad to see this bring added to the repo!  Here are some things that ideally we can fix before import, that would be great.  If you can't, please add a backlog item for each to address later.

**Comment by mmacy** [comment]
@Dickson-Mwendia I just merged #23 from @henrymbuguakiarie but now wondering if we need to implement the review changes here into that. Maybe I should have merged this one?

Not sure - can you two sort it out and let me know?

**Pull request closed without merging**